src/app/AttributePathExpandIterator.cpp

```diff
@@ -93,13 +94,24 @@ bool AttributePathExpandIterator::AdvanceOutputPath()
     }
 }
 
-bool AttributePathExpandIterator::Next(ConcreteAttributePath & path)
+bool AttributePathExpandIterator::Next(ConcreteAttributePath & path, std::optional<DataModel::AttributeEntry> * entry)
 {
     while (mPosition.mAttributePath != nullptr)
     {
         if (AdvanceOutputPath())
         {
             path = mPosition.mOutputPath;
+            if (entry != nullptr)
+            {
+                if (mPosition.mEntry)
+                {
+                    entry->emplace(*mPosition.mEntry);
+                }
+                else
+                {
+                    entry->reset();
+                }
```

Comment:
This seems like a complicated way of writing `*entry = mPosition.mEntry`, no?

Reply:
Code was updated in the mean time.